src/account/sync/input_selection.rs

```diff
@@ -1,11 +1,19 @@
+use crate::address::Address;
 use rand::{thread_rng, Rng};
 
-pub fn select_input(target: u64, available_utxos: &mut Vec<u64>) -> crate::Result<Vec<u64>> {
-    if target > available_utxos.iter().sum::<u64>() {
+pub fn select_input(
+    target: u64,
+    available_utxos: &mut Vec<Address>,
```

Comment:
```suggestion
    available_utxos: &mut [Address],
```

I would prefer slice since `Vec` has `Deref` trait and could take advantage of deref coercion. And other type that has slice as their Deref output could also benefit from it.